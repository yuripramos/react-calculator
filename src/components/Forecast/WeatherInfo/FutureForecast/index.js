import FutureForecast from "./FutureForecast";

export default FutureForecast;
