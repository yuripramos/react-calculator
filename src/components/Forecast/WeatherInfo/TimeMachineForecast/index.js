import TimeMachineForecast from "./TimeMachineForecast";

export default TimeMachineForecast;
