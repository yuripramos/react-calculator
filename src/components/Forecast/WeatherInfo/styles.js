import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 70%;
  display: inline-block;
`;
export const InnerContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  display:inline-flex;
`;

export const City = styled.div`
  text-align: left;
  border-bottom: solid thin white;
  text-transform: uppercase;
  color: #e8e8e8;
  color: rgba(255, 255, 255, 0.7);
  font-family: "RobotoSlab Thin";
  font-size: ${rem(22)};
  padding: 0 ${rem(15)};
`;

export const Icon = styled.div`
  width: 100%;
  height: ${rem(150)};
`;

export const Temperature = styled.h1`
  font-size: 4.5em;
  font-weight: 300;
  line-height: 0.75;
  font-family: "RobotoSlab Regular";
  margin-top: ${rem(20)};
  width: 100%;
  ${({ tiny }) =>
    tiny &&
    css`
    margin-top: ${rem(5)};
      font-size: 1em;
  `};
`;

export const TinySpecs = styled.span`
    color: white;
    font-weight: 100;
    line-height: 1.1;
    letter-spacing: 0.025em;
    margin: 0;
    padding: 0;
    font-family: "OpenSans Thin";
    width: 100%;
    margin: ${rem(10)} 0;
    ${({ tiny }) =>
      tiny &&
      css`
        font-size: .9em;
    `};
    :nth-child(2n) {
      margin: ${rem(2)} 0;
    }
  `;

export const MainContentWrapper = styled.div`
  display: inline-flex;
  justify-content: center;
  align-content: flex-start;
  width: 65%;
  flex-wrap: wrap;
`;

export const SidebarContentWrapper = styled.div`
  display: flex;
  width: 35%;
  flex-wrap: wrap;
  text-align: right;
  height: ${rem(180)};
`;
