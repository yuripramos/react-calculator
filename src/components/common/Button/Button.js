import React from "react";
import { bool, node, string, func, number } from "prop-types";

import { ButtonTag, RouterLinkTag } from "./styles";
import LocalLoading from "../../common/LocalLoading";
function Button({
  children,
  onClick,
  type,
  disabled,
  width,
  opacity,
  to,
  isCallToAction,
  loading
}) {
  if (to) {
    return (
      <RouterLinkTag
        width={width}
        onClick={onClick}
        isCallToAction={isCallToAction}
        to={to}
        disabled={disabled}
        loading={loading.toString()}
      >
        {loading ? <LocalLoading /> : children}
      </RouterLinkTag>
    );
  }
  return (
    <ButtonTag
      width={width}
      type={type}
      disabled={disabled}
      onClick={onClick}
      opacity={opacity}
      loading={loading}
    >
      {children}
    </ButtonTag>
  );
}

Button.displayName = "Button";

Button.defaultProps = {
  type: "button",
  disabled: false,
  width: null,
  onClick: null,
  className: null,
  to: null,
  loading: false,
};

Button.propTypes = {
  children: node.isRequired, // eslint-disable-line
  isCallToAction: bool,
  disabled: bool,
  type: string,
  width: string,
  onClick: func,
  opacity: number,
  loading: bool
};

export default Button;
