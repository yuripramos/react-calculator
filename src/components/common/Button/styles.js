import React from "react";
import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { Link as RouterLink } from "react-router-dom";
import { white, green, green20, black30 } from "../../../styles/settings";

export const ButtonTag = styled.button`
  font-family: "RobotoSlab Regular";
  border-radius: ${rem(4)};
  border: 0;
  cursor: pointer;
  display: inline-block;
  font-size: ${rem(12)};
  letter-spacing: ${rem(0.5)};
  line-height: ${rem(45)};
  outline: none;
  padding: ${rem(5)};
  position: relative;
  text-align: center;
  text-decoration: none;
  color: ${white};
  transition: 0.2s ease-in-out color, 0.2s ease-in-out background,
    0.2s ease-in-out box-shadow, 0.2s ease-in-out border, 0.1s transform;
  white-space: nowrap;
  width: ${props => (props.width ? rem(props.width) : "100%")};
  user-select: none;
  opacity: ${props => (props.opacity && props.disabled ? props.opacity : 1)};
  background: transparent;
  transform: translateY(15px);

  ${media.md(css`
    width: ${props => (props.width ? rem(props.width) : "auto")};
  `)};

  ${media.lg(css`
    font-size: ${rem(13)};
    line-height: ${rem(48)};
  `)};

  :active {
    transform: translateY(${rem(1)});
  }
  :hover {
    background: ${white};
    color: ${black30};
    transform: translateY(20px);
  }

  ${({ isCallToAction }) =>
    isCallToAction &&
    css`
      background-color: ${green};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(100, 100, 100, 0.3);
      color: ${white};
      :hover {
        background: ${green20};
      }
    `};
`;

export const RouterLinkTag = ButtonTag.withComponent(
  ({ isCallToAction, ...props }) => <RouterLink {...props} /> // eslint-disable-line
);
