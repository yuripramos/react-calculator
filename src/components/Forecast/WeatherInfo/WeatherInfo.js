import React, { Component } from "react";
import { string, arrayOf, number, bool, shape, func } from "prop-types";

import {
  ContentWrapper,
  Temperature,
  City,
  TinySpecs,
  Icon,
  MainContentWrapper,
  SidebarContentWrapper,
  InnerContentWrapper
} from "./styles";
import Button from "../../common/Button";
import IconGenerator from "../../common/Icon";
import FutureForecast from "./FutureForecast";
import TimeMachineForecast from "./TimeMachineForecast";
import { Container, Row, Column } from "../../../styles/grid";

class WeatherInfo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false
    };
    this.handleConfirmButton = this.handleConfirmButton.bind(this);
  }

  async handleConfirmButton() {
    const {
      isTimeMachineActive,
      latitude,
      longitude,
      toggleForecast,
      getForecastTimeMachine,
    } = this.props;

    this.setState({
      isLoading: true
    });
    !isTimeMachineActive && (await getForecastTimeMachine(latitude, longitude));
    isTimeMachineActive && toggleForecast();
    this.setState({
      isLoading: false
    });
  }
  render() {
    const {
      city,
      temperature,
      day,
      wind,
      humidity,
      icon,
      summary,
      apparentTemperature,
      nextWeek,
      isTimeMachineActive,
      forecastTimeMachine,
      clearSearch
    } = this.props;
    const { isLoading } = this.state;
    return (
      <ContentWrapper>
        <Container>
          <Row>
            <Column>
              <City>{city}</City>
              <InnerContentWrapper>
                <MainContentWrapper>
                  <Icon>
                    <IconGenerator
                      name={icon}
                      width={"170px"}
                      height={"170px"}
                    />
                  </Icon>
                  <Temperature> {temperature} </Temperature>
                  <TinySpecs tiny>
                    Apparent temperature
                    <Temperature tiny>{apparentTemperature}</Temperature>
                  </TinySpecs>
                </MainContentWrapper>
                <SidebarContentWrapper>
                  <TinySpecs>{day}</TinySpecs>
                  <TinySpecs>{summary}</TinySpecs>
                  <TinySpecs> Wind: {wind} </TinySpecs>
                  <TinySpecs> Humidity: {humidity} </TinySpecs>
                </SidebarContentWrapper>
              </InnerContentWrapper>
              {isTimeMachineActive ? (
                <TimeMachineForecast
                  forecast={forecastTimeMachine}
                  isTimeMachineActive={isTimeMachineActive}
                  clearSearch={clearSearch}
                />
              ) : (
                <FutureForecast forecast={nextWeek} />
              )}
              <Button
                width={"180"}
                to={{
                  pathname: "/forecast",
                  search: `?time-machine-request&search=${city}`
                }}
                loading={isLoading}
                onClick={this.handleConfirmButton}
              >
                {isTimeMachineActive
                  ? "7-days future forecast"
                  : "click here to 30-day past"}
              </Button>
            </Column>
          </Row>
        </Container>
      </ContentWrapper>
    );
  }
}

WeatherInfo.defaultProps = {};

WeatherInfo.propTypes = {
  city: string,
  temperature: string,
  day: string,
  wind: string,
  humidity: string,
  icon: string,
  summary: string,
  clearSearch: func,
  nextWeek: arrayOf(
    shape({
      time: number,
      icon: string,
      temperatureHigh: number,
      temperatureLow: number
    })
  ),
  apparentTemperature: string,
  isTimeMachineActive: bool,
  forecastTimeMachine: arrayOf(shape)
};

export default WeatherInfo;
