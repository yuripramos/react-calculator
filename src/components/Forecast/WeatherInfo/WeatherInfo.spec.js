import React from "react";
import { shallow } from "enzyme";

import WeatherInfo from "./WeatherInfo";
import Button from "../../common/Button";
import { spy } from "sinon";
let props;

beforeEach(() => {
  props = {
    city: "one city",
    temperature: 33,
    day: "21 January, 2019",
    wind: 3,
    humidity: 23,
    icon: "clear-day",
    summary: "Amazing day, go to the beach",
    apparentTemperature: 30,
    nextWeek: [
      {
        temperatureHigh: 35,
        temperatureLow: 18,
        time: 2312321,
        icon: "clear-night"
      },
      {
        temperatureHigh: 25,
        temperatureLow: 8,
        time: 2314441,
        icon: "clear-day"
      },
      {
        temperatureHigh: 25,
        temperatureLow: 8,
        time: 231111141,
        icon: "clear-day"
      }
    ],
    isTimeMachine: false,
    forecastTimeMachine: [
      {
        currently: {
          time: 214124,
          icon: "fog",
          temperature: 20
        }
      },
      {
        currently: {
          time: 123124,
          icon: "snow",
          temperature: -10
        }
      }
    ]


  };
});

describe("FutureForecast Component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<WeatherInfo {...props} />)).toMatchSnapshot();
  });
  it("should call toggleforecast when timeMachine is active", () => {
    spy(WeatherInfo.prototype, "handleConfirmButton");

    const wrapper = shallow(<WeatherInfo {...props} />);
    wrapper.find(Button).simulate("click");
    wrapper.setProps({ isTimeMachineActive: true });
    wrapper.update();
    expect(WeatherInfo.prototype.handleConfirmButton.calledOnce);

    WeatherInfo.prototype.handleConfirmButton.restore();
  });
});
