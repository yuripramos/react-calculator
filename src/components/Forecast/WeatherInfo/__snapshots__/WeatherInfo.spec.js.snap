// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FutureForecast Component should match snapshot with props 1`] = `
<styled.div>
  <Container>
    <Row>
      <Column>
        <styled.div>
          one city
        </styled.div>
        <styled.div>
          <styled.div>
            <styled.div>
              <IconGenerator
                height="170px"
                name="clear-day"
                width="170px"
              />
            </styled.div>
            <styled.h1>
               
              33
               
            </styled.h1>
            <styled.span
              tiny={true}
            >
              Apparent temperature
              <styled.h1
                tiny={true}
              >
                30
              </styled.h1>
            </styled.span>
          </styled.div>
          <styled.div>
            <styled.span>
              21 January, 2019
            </styled.span>
            <styled.span>
              Amazing day, go to the beach
            </styled.span>
            <styled.span>
               Wind: 
              3
               
            </styled.span>
            <styled.span>
               Humidity: 
              23
               
            </styled.span>
          </styled.div>
        </styled.div>
        <FutureForecast
          forecast={
            Array [
              Object {
                "icon": "clear-night",
                "temperatureHigh": 35,
                "temperatureLow": 18,
                "time": 2312321,
              },
              Object {
                "icon": "clear-day",
                "temperatureHigh": 25,
                "temperatureLow": 8,
                "time": 2314441,
              },
              Object {
                "icon": "clear-day",
                "temperatureHigh": 25,
                "temperatureLow": 8,
                "time": 231111141,
              },
            ]
          }
        />
        <Button
          className={null}
          disabled={false}
          loading={false}
          onClick={[Function]}
          to={
            Object {
              "pathname": "/forecast",
              "search": "?time-machine-request&search=one city",
            }
          }
          type="button"
          width="180"
        >
          click here to 30-day past
        </Button>
      </Column>
    </Row>
  </Container>
</styled.div>
`;
